src/main/java/org/spongepowered/common/event/SpongeEventManager.java

```diff
@@ -140,6 +141,12 @@ public void registerListener(PluginContainer plugin, Object listenerObject) {
         checkNotNull(plugin, "plugin");
         checkNotNull(listenerObject, "listener");
 
+        if (this.registeredListeners.contains(listenerObject)) {
+            SpongeImpl.getLogger().warn("Plugin {} attempted to register an already registered listener ({})", plugin.getId(),
+                    listenerObject.getClass().getName());
```

Comment:
Would be nice to dump the stack trace as well. I have a private Bukkit plugin which (I believe) is registering a certain listener twice, and I don't know where the second registration is.